teps/0018-default-bundle.md

```diff
@@ -0,0 +1,165 @@
+---
+title: Allow a Run to Specify a Default Bundle
+authors:
+  - "@coryrc"
+creation-date: 2020-09-15
+last-updated: 2020-09-15
+status: in pr review
+---
+
+# TEP-0018 Allow a Run to Specify a Default Bundle
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Observability](#observability)
+  - [New Variables](#new-variables)
+  - [Explicit Bundles from in-repo Resources](#explicit-bundles-from-in-repo-resources)
+- [Test Plan](#test-plan)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+In a Pipeline, a Task can be referenced by name. This name is currently looked
+up in the current namespace where the Pipeline and PipelineRun exist. It should
+be possible for a PipelineRun or TaskRun to specify a bundle to be searched for
+a match instead.
+
+## Motivation
+
+When CI is defined in-repo, some method must be provided to test Tekton objects
+during presubmit without altering Tasks being used by others. Additionally
+support multiple repositories and branches in a single namespace without the
+possibility of name conflict.
+
+### Goals
+
+Provide a different place to lookup Tekton objects when referenced solely by name.
+
+### Non-Goals
+
+- Override explicitly-labeled referenced Tasks (i.e. if they have taskRef.bundle).
+  This means further work is needed for presubmit testing of bundles generated
+  from resources defined in the repo-under-test. [more](#explicit-bundles-from-in-repo-resources)
+
+## Requirements
+
+## Proposal
+
+Create a new key for PipelineRun and TaskRun called `spec.lookup.bundle`; use it
```

Comment:
This is an implementation detail but since you mention the API, something about a `lookup` field feels like the wrong tense?

One thing that other fields in Tekton have is a "default" field. Eg, params have a value and a default. Perhaps a `PipelineRun`, and `TaskRun` object need a `default` block for things like this? Something like `spec.defaults.bundle`?

Reply:
Sorry, I am not sufficiently knowledgeable about k8s to judge names! You all decide and I'll write it down!